### ISSUE TITLE
Change gcsfuse solution to point to allow access to all buckets

### ISSUE THREAD

**Issue by mrocklin** [pangeo-data/pangeo#93]
Currently the gcsfs fuse solution points to a particular bucket

    $ pip install gcsfs
    $ gcsfuse pangeo-data /gcs

According to users at UK Met it's actually quite handy if the `/gcs` mount (or in their case `/s3`) points to the entire global object store.  Obviously operations like `ls` won't work here, but users can `cd` into a particular bucket that they know, or refer to any object in any bucket if they know the bucket name ahead of time.

cc @martindurant

**Comment by martindurant** [comment]
I agree that this could be useful, but I am not sure whether it is technically possible to `cd` into a directory that has not been previously listed. That might depend on the exact navigation pattern and maybe the shell implementation.

**Comment by mrocklin** [comment]
I'm encouraged by the fact that UK Met has been able to do this with S3

**Comment by jhamman** [comment]
I think this would be worth trying, for both S3 and GCS. There is a lot of public data on GCS and S3 that we cannot currently access.

**Comment by martindurant** [comment]
I hadn't thought to try it, but actually you can provide the bucket name as an empty string, and it works just fine
```
$ gcsfuse  "" temp/gcs
..
$ ls temp/gcs
artifacts.pangeo-181919.appspot.com pangeo-data                         zarr-test
pangeo                              pangeo-data-private
$ ls temp/gcs/pangeo
mrocklin    pangeo-data test
..
In [1]: d = xr.open_dataset('temp/gcs/pangeo-data/newmann-met-ensemble-netcdf/conus_ens_001.nc')
```
and you can use this for any public bucket (although they won't show up when listing the top directory).
